tracehandler/tracehandler.go

```diff
@@ -132,19 +126,14 @@ func (m *traceHandler) HandleTrace(bpfTrace *host.Trace) {
 	defer m.traceProcessor.SymbolizationComplete(bpfTrace.KTime)
 	timestamp := libpf.UnixTime64(m.times.BootTimeUnixNano() + int64(bpfTrace.KTime))
 
-	meta, err := m.containerMetadataHandler.GetContainerMetadata(bpfTrace.PID)
-	if err != nil {
-		log.Warnf("Failed to determine container info for trace: %v", err)
-	}
-
 	if !m.reporter.SupportsReportTraceEvent() {
 		// Fast path: if the trace is already known remotely, we just send a counter update.
 		postConvHash, traceKnown := m.bpfTraceCache.Get(bpfTrace.Hash)
 		if traceKnown {
 			m.bpfTraceCacheHit++
 			svcName := m.traceProcessor.MaybeNotifyAPMAgent(bpfTrace, postConvHash, 1)
 			m.reporter.ReportCountForTrace(postConvHash, timestamp, 1,
-				bpfTrace.Comm, meta.PodName, meta.ContainerName, svcName)
+				bpfTrace.Comm, "", "", svcName)
```

Comment:
I think this only works out if the PID is used here.

Reply:
I'd be fine with that, just in the interest of keeping consumers working regardless of the commit it wouldn't hurt to do it here either. No strong opinions thoughi

Reply:
One option would be to allow adding arbitrary metadata from pluggable user-code - either before passing the trace + metadata to the reporter or as part of the reporter (I'd prefer the former).
The current agent can then add some metadata as an example, without the requirement to pull in large dependencies (aws, k8s, ...) on the library level. This allows much smaller binaries for users who don't require to detect cloud environments.

Reply:
Yeah, my understanding is (but it's rather limited so we should get thoughts from others) there's only a minimal set of metadata the `otel-profiling-agent` needs to add for the otel collector to be able to add any additional metadata. This way the `otel-profiling-agent` can be relatively lightweight in terms of dependencies for metadata. I don't know the exact mechanics though, we should ask distributed tracing folks how this is intended to work.